### PR TITLE
Blocks: Make webpack.config.extensions.js extend calypso-build's

### PR DESCRIPTION
**Pull request by ockham**
#### Changes proposed in this Pull Request:

- Bump `@automattic/calypso-build` to `1.0.0-alpha.2`, which includes a basline `webpack.config.js`.
- Base `webpack.config.extensions.js` on that.

#### Testing instructions:

```
yarn distclean
yarn build-extensions
```

Verify that `index.json` is copied to `_inc/blocks/`, that there are `.deps.json` files listing dependencies, and that blocks are still working.

#### Proposed changelog entry for your changes:

Blocks: Make `webpack.config.extensions.js` extend `@automattic/calypso-build`'s

### CODE REVIEW

File: webpack.config.extensions.js
Comment (sirreal) [inline webpack.config.extensions.js]:
```suggestion
		entry: {
		'index.json': 'extensions/setup/index.json',
```

**Review by sirreal — COMMENTED** [review]
Noting some thoughts we had while on a call.

File: webpack.config.extensions.js
Comment (ockham) [reply webpack.config.extensions.js]: This works _almost_ -- webpack appends `.js` to `index.json` :sweat_smile:

Comment (ockham) [reply webpack.config.extensions.js]: (And it does weird webpack things to it)

File: webpack.config.extensions.js
Comment (sirreal) [inline webpack.config.extensions.js]: Generally good style, but we've just produced the webpackConfig, we "own" it. Also, merging into a new object is slightly misleading because `merge` is deep, so the object contained in `webpackConfig` will be mutated anyways.

We could merge right where we create the config: `module.exports = _.merge( getBaseWebpackConfig(…), { … } )`

```suggestion
module.exports = _.merge( webpackConfig, {
```

**Review by sirreal — COMMENTED** [review]
One note, code looks good, I'll test…

**Review by sirreal — DISMISSED** [review]
Seems to work well, I've tested normal and production bundles 👍

File: webpack.config.extensions.js
Comment (ockham) [reply webpack.config.extensions.js]: > Generally good style, but we've just produced the webpackConfig, we "own" it. Also, merging into a new object is slightly misleading because `merge` is deep, so the object contained in `webpackConfig` will be mutated anyways.

:thinking: This has me confused. I'm reading 'deep' merge as 'recursively merging properties from source objects into the destination one' but I didn't expect source objects to be mutated? See https://lodash.com/docs/4.17.11#merge

> We could merge right where we create the config: `module.exports = _.merge( getBaseWebpackConfig(…), { … } )`

Yeah, that makes sense :+1: I was avoiding it because of nesting levels, but it shouldn't be too bad since there aren't too many LOCs involved here.

Comment (ockham) [reply webpack.config.extensions.js]: Hmm it looks a bit more distracting with the `getBaseWebpackConfig()` call inlined after all :sweat_smile:

File: webpack.config.extensions.js
Comment (simison) [inline webpack.config.extensions.js]: Would it work to check for `view.scss` if `view.js` doesn't exist? Would that avoid creating empty `view.js` files when only CSS is needed?

File: webpack.config.extensions.js
Comment (sirreal) [reply webpack.config.extensions.js]: Apparently my interpretation was off, it _doesn't_ seem to mutate the contents:

```js
const { merge } = require('lodash')
const a = { b: { orig: 'b' } }
const toMerge = { b: { new: 'b' } }
const a_ = merge( {}, a, toMerge )
a.b  // { orig: 'b' }
a_.b // { orig: 'b', new: 'b' }
```

File: webpack.config.extensions.js
Comment (simison) [inline webpack.config.extensions.js]: I expected not needing to copy this file anymore? Re https://github.com/Automattic/jetpack/pull/11801

File: webpack.config.extensions.js
Comment (sirreal) [reply webpack.config.extensions.js]: Please, leave this. It's good as-is 👍

File: webpack.config.extensions.js
Comment (ockham) [reply webpack.config.extensions.js]: Jon and I discussed this at length during our call yesterday. While we still want to remove it eventually, getting it right is complex, and I definitely wanted to unblock this PR -- so keeping it for the time being made the most sense :slightly_smiling_face:

Comment (simison) [reply webpack.config.extensions.js]: Makes sense 👍  I just wasn't fully aware at which order things are happening, as I remembered seeing a PR about it in the past. :-)

File: webpack.config.extensions.js
Comment (ockham) [reply webpack.config.extensions.js]: Not sure I'm following, sorry :sweat_smile: The check on the next line makes sure that we're not creating an empty `view.js` in the bundle if there is no source file with that name. Did you mean something else?

Comment (simison) [reply webpack.config.extensions.js]: I meant that in situation where we want to have `view.css` but don't have any JS, we currently need to create both files, where `view.js` really just is:

```js
import 'view.scss';
```

I'm suggesting that folks could not create `view.js` and we'd go ahead and still pick `view.scss` if it exists and build `view.css`.

Comment (ockham) [reply webpack.config.extensions.js]: It does make sense! :smile: 

I agree that the dummy JS is kinda annoying, but TBH, I'd rather tackle that separately. It's arguably a somewhat different concern from what this PR does, and might require enough thought to warrant a separate PR :slightly_smiling_face:

File: package.json
Comment (simison) [inline package.json]: Already done in https://github.com/Automattic/jetpack/pull/11941

File: package.json
Comment (simison) [inline package.json]: is this `@automattic/calypso-color-schemes` anymore used in Jetpack directly as it's now also [dependency in calypso-build](https://github.com/Automattic/jetpack/pull/11802/files#diff-8ee2343978836a779dc9f8d6b794c3b2R10)?

(asked the same [just elsewhere](https://github.com/Automattic/jetpack/pull/11941/files#r272532005) accidentally)

Comment (ockham) [reply package.json]: Good spot! I'll remove.

File: package.json
Comment (ockham) [reply package.json]: Fixed by a rebase.

**Comment by jeherve** [comment]
This is going to need a rebase I am afraid.

**Comment by ockham** [comment]
Rebased. I also need to bump `calypso-build` tho, this will take a few.

**Comment by ockham** [comment]
> Rebased. I also need to bump `calypso-build` tho, this will take a few.

Done. should be ready for review again.

File: package.json
Comment (simison) [reply package.json]: Just noting that actually we might want to bring this dependency back at some point, since looks like it would be useful for Calypsoify module:

https://github.com/Automattic/jetpack/blob/4884963a1049378290f0b35584084f251075b168/modules/calypsoify/_colors.scss

**Comment by simison** [comment]
Rebased with changes in `package.json`

**Review by simison — APPROVED** [review]
Looks good and tests well.

Tested with blocks and Jetpack dashboard using `yarn build` and `yarn build-extensions`. Json files are there, too. 👍

**Review by oskosk — APPROVED** [review]

**Comment by ockham** [comment]
Thanks Osk!

I'll force wpcom status to Green since I need to update a bunch of dependencies on WP.com in a separate phab diff before I can rebase D26294-code. I'm on it, but I don't want to block this PR anymore (since I'm worried I'd otherwise lose Crew approval again).

**Comment by lezama** [comment]
this is beautiful

**Pull request merged**
